### PR TITLE
ci: add codeql action

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Add codeql action to run on merge to `master`, PRs and periodically

Test run: https://github.com/aramase/secrets-store-csi-driver-provider-azure/runs/8259861468?check_suite_focus=true

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
```
Comment (sozercan) [inline .github/workflows/codeql.yaml:7]: is it going to add too much extra time to PR validation?

Comment (aramase) [reply .github/workflows/codeql.yaml:7]: Validated it here: https://github.com/aramase/secrets-store-csi-driver-provider-azure/runs/8259861468?check_suite_focus=true and it only takes 3m40s

File: .github/workflows/codeql.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    branches:
+    - master
+  schedule:
+    - cron: "0 7 * * 1" # Mondays at 7:00 AM
```
Comment (nilekhc) [inline .github/workflows/codeql.yaml:11]: Is this UTC or PST?

File: .github/workflows/codeql.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@b398f525a5587552e573b247ac661067fafa920b
+        with:
+          languages: go
+
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@b398f525a5587552e573b247ac661067fafa920b
+
+      - name: Perform CodeQL Analysis
```
Comment (nilekhc) [inline .github/workflows/codeql.yaml:34]: Can we visualize the output of this step? The action says it's uploading data.

File: .github/workflows/codeql.yaml (hunk 2 above)
Comment (aramase) [reply .github/workflows/codeql.yaml:11]: It's UTC

File: .github/workflows/codeql.yaml (hunk 3 above)
Comment (aramase) [reply .github/workflows/codeql.yaml:34]: Any alerts from that run will be available in [Code Scanning](https://github.com/Azure/secrets-store-csi-driver-provider-azure/security/code-scanning)

File: .github/workflows/codeql.yaml (hunk 2 above)
Comment (nilekhc) [reply .github/workflows/codeql.yaml:11]: Do we want to move to PST morning?

Comment (aramase) [reply .github/workflows/codeql.yaml:11]: I've no strong opinion on that. If you would like me to change it, I can do that.

Comment (nilekhc) [reply .github/workflows/codeql.yaml:11]: Let's do it!

Comment (aramase) [reply .github/workflows/codeql.yaml]: done!

**Review by nilekhc — APPROVED** [review]

**Pull request merged**